src/timezones/local.jl

```diff
@@ -0,0 +1,237 @@
+# Determine the local systems timezone
+# Based upon Python's tzlocal https://pypi.python.org/pypi/tzlocal
+
+function get_localzone()
+    @windows? _get_localzone_windows():(
+        @osx? _get_localzone_mac():(
+            _get_localzone_unix()
+        )
+    )
+end
+
+function _get_localzone_mac()
+    zone = readall([`systemsetup -gettimezone`]...)
+    if contains(zone, "Time Zone: ")
+        zone = strip(replace(zone, "Time Zone: ", ""))
+    else
+        zone = readlink(["/etc/localtime"]...)
+        # link will be something like /usr/share/zoneinfo/America/Winnipeg
+        zone = match(r"(?<=zoneinfo/).*$", zone).match
+    end
+    return TimeZone(zone)
+end
+
+function _get_localzone_unix()
+    validnames = timezone_names()
+
+    # Try getting the time zone "TZ" environment variable
+    # http://linux.die.net/man/3/tzset
+    zone = Nullable{AbstractString}(get(ENV, "TZ", nothing))
+    if !isnull(zone)
+        zone_str = get(zone)
+        if startswith(zone_str,':')
+            zone_str = zone_str[2:end]
+        end
+        zone_str in validnames && return TimeZone(zone_str)
+
+        # Read the tzfile to get the timezone
+        try
+            return build_tzinfo("local", zone_str)
+        catch
+            error("Failed to resolve local timezone from \"TZ\" environment variable.")
+        end
+    end
+
+    # Look for distribution specific configuration files
+    # that contain the timezone name.
+
+    filename = "/etc/timezone"
+    if isfile([filename]...)
+        file = open([filename]...)
```

Comment:
The `try finally` patter was being used since it appears to work better with mocking. But overall I agree.